app/controllers/staff/activity_forms_controller.rb

```diff
@@ -197,4 +198,16 @@ def country_to_region_mapping
     yaml = YAML.safe_load(File.read("#{Rails.root}/vendor/data/codelists/BEIS/country_to_region_mapping.yml"))
     yaml["data"]
   end
+
+  def add_transparency_identifier
+    return if @activity.identifier.nil?
+    return if @activity.transparency_identifier.present?
+    @activity.update(transparency_identifier: iati_identifier)
+  end
+
+  def iati_identifier
```

Comment:
I wonder if we could move this method to the activity model and move across the existing unit tests from `ActivityXmlPresenter`. This would keep the extra unit test coverage and allow us to remove this heavier logic from the controller which I think are both additional improvements.